src/pymor/bindings/dunegdt.py

```diff
@@ -0,0 +1,418 @@
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.core.config import config
+
+
+if config.HAVE_DUNEGDT:
+
+    import numpy as np
+    from functools import partial
+    from collections import OrderedDict
+    import os
+    import subprocess
+    import tempfile
+    from tempfile import mkstemp
+    from matplotlib import pyplot as plt
+
+    from dune.xt.common.vtk.plot import plot as k3d_plot
+    from dune.gdt import DiscreteFunction
+
+    from pymor.core.base import ImmutableObject
+    from pymor.discretizers.builtin.grids.oned import OnedGrid
+    from pymor.discretizers.builtin.gui.visualizers import OnedVisualizer
+    from pymor.operators.constructions import ZeroOperator
+    from pymor.operators.list import ListVectorArrayOperatorBase
+    from pymor.vectorarrays.interface import _create_random_values
+    from pymor.vectorarrays.list import ListVectorArray, Vector, ListVectorSpace
+    from pymor.vectorarrays.numpy import NumpyVectorSpace
+
+
+    class DuneXTVector(Vector):
+        """Wraps a vector from dune-xt to make it usable with ListVectorArray."""
+
+        def __init__(self, impl):
+            self.impl = impl
+
+        # @property
+        # def data(self):
+            # return np.frombuffer(self.impl)
+
+        def copy(self, deep=False):
+            return DuneXTVector(self.impl.copy(deep))
+
+        def scal(self, alpha):
+            self.impl.scal(alpha)
+
+        def axpy(self, alpha, x):
+            self.impl.axpy(alpha, x.impl)
+
+        def inner(self, other):
+            return self.impl.dot(other.impl)
+
+        def norm(self):
+            return self.impl.l2_norm()
+
+        def norm2(self):
+            return self.impl.l2_norm() ** 2
+
+        def sup_norm(self):
+            return self.impl.sup_norm()
+
+        def dofs(self, dof_indices):
+            impl = self.impl
+            return np.array([impl[i] for i in dof_indices])
+
+        def amax(self):
+            _amax = self.impl.amax()
+            return _amax[0], _amax[1]
+
+        def __add__(self, other):
+            return DuneXTVector(self.impl + other.impl)
+
+        def __iadd__(self, other):
+            self.impl += other.impl
+            return self
+
+        __radd__ = __add__
+
+        def __sub__(self, other):
+            return DuneXTVector(self.impl - other.impl)
+
+        def __isub__(self, other):
+            self.impl -= other.impl
+            return self
+
+        def __mul__(self, other):
+            return DuneXTVector(self.impl * other)
+
+        def __imul__(self, other):
+            self.impl *= other
+            return self
+
+        def __neg__(self):
+            return self * (-1)
+
+        def to_numpy(self, ensure_copy=False):
+            return np.array(self.impl, copy=ensure_copy)
+
+
+    class DuneXTVectorSpace(ListVectorSpace):
+
+        def __init__(self, vector_type, dim, id='STATE'):
```

Comment:
`vector_type` is LA backend dependent? Docs?

Reply:
is now set to a suitable default in #1383 which then also makes the intention clearer (I hope)